crypto/ex_data.c

```diff
@@ -380,14 +380,17 @@ int CRYPTO_alloc_ex_data(int class_index, void *obj, CRYPTO_EX_DATA *ad,
         return 1;
 
     ip = get_and_lock(class_index);
+    if (ip == NULL)
+        return 0;
     f = sk_EX_CALLBACK_value(ip->meth, idx);
     CRYPTO_THREAD_unlock(ex_data_lock);
 
     /*
      * This should end up calling CRYPTO_set_ex_data(), which allocates
      * everything necessary to support placing the new data in the right spot.
      */
-    f->new_func(obj, curval, ad, idx, f->argl, f->argp);
+    if (f->new_func)
+        f->new_func(obj, curval, ad, idx, f->argl, f->argp);
 
     return 1;
```

Comment:
It should *not* return 1 if allocation failed for any reason.  That include `f->new_func` being NULL

Reply:
I was also thinking  that `new_func` returns something.
Until the compilation fails, because its signature is just : 
```
typedef void CRYPTO_EX_new (void *parent, void *ptr, CRYPTO_EX_DATA *ad,
                           int idx, long argl, void *argp);
```
It is a kind of constructor/initializer, that's do not expected failure.

Reply:
`f->new_func` being NULL is a fully possible value , nothing prevent it.
 and code is guard against.

Reply:
I think what @levitte is saying is that you need something like
```
   if  (f->new_func == NULL)
      return 0
   f->new_func(obj, curval, ad, idx, f->argl, f->argp);
```

Reply:
@kroeckx , Ok, I read this comment too quickly. I rework my fix in this way. Thank for the answer.
CI's  seems OK .